src/main/kotlin/com/zepben/evolve/database/sqlite/DatabaseReader.kt

```diff
@@ -96,7 +96,7 @@ class DatabaseReader @JvmOverloads constructor(
             return false
         }
 
-        return status and postLoad(networkService)
+        return status && postLoad(networkService)
```

Comment:
The database load was designed to do everything, even if there are failures, hence using `and` rather than `&&`. This was done to allow the log to be filled with as many errors as possible in a single pass. It is painful to load and be told a single error, which you fix and reload to find out it still fails for other previously hidden reasons.

I am happy for the post load to be "skipped" if there are load errors as often they will impact the processes that are run. I typically just stop the load when I see an error while testing a migrator if I don't want to wait.

This is only an issue only when a migrator is broken (not producing a valid database), so if the cost of waiting for the phase/direction assignment is considered an issue, at least call the proper clean up code for a clean exit (as a side effect of allowing this to short circuit, you are no longer calling `closeConnection`, which probably makes next to no difference as the process will exit anyway).

Reply:
Should also have mentioned that the `and` was used to be consistent with the service load functions that read much better as `status = status and ...` in repeated lines rather than `status = ... && status` due to the different line lengths (see [NetworkServiceReader](https://github.com/zepben/evolve-sdk-jvm/blob/main/src/main/kotlin/com/zepben/evolve/database/sqlite/readers/NetworkServiceReader.kt)).

Reply:
I've made the changes. Up to you if you think it's worthy of merging; I am not fussed either way.

In my mind they are two distinct phases of the start up process and you'd never bother investigating/fixing errors in the second phase until you'd resolved all of the errors in the database load, because as you say they may be impacting the post load processes.